cider-dap/src/adapter.rs

```diff
@@ -1,25 +1,48 @@
-use dap::types::{Breakpoint, Source, SourceBreakpoint, Thread};
-use std::fs::File;
+use crate::error::AdapterResult;
+use dap::types::{Breakpoint, Source, SourceBreakpoint, StackFrame, Thread};
+use interp::debugger::source::structures::NewSourceMap;
+use interp::debugger::Debugger;
+use std::collections::HashMap;
+use std::path::PathBuf;
+
 pub struct MyAdapter {
     #[allow(dead_code)]
-    file: File,
-    breakpoints: Vec<(Source, i64)>, //This field is a placeholder
+    debugger: Debugger,
     break_count: Counter,
     thread_count: Counter,
-    threads: Vec<Thread>, //This field is a placeholder
+    stack_count: Counter,
+    breakpoints: Vec<(Source, i64)>, // This field is a placeholder
+    stack_frames: Vec<StackFrame>,   // This field is a placeholder
+    threads: Vec<Thread>,            // This field is a placeholder
+    source: String,
+    ids: NewSourceMap,
 }
 
+// New metadata in interp/debugger
+
 impl MyAdapter {
-    pub fn new(file: File) -> Self {
-        MyAdapter {
-            file,
-            breakpoints: Vec::new(),
+    // Look at Rust File implementation
+    // Pass in path, easier
+    // Change to take in the file path
+    // Create open file function
+    pub fn new(path: &str) -> AdapterResult<Self> {
+        Ok(MyAdapter {
+            debugger: Debugger::from_file(
+                &PathBuf::from(path),
+                // Hard code for now, change path as necessary
+                &PathBuf::from("/home/elias/calyx/calyx-stdlib"),
+            )
+            .unwrap(),
```

Comment:
Great! I think the thing to do here is to make the stb_lib path a configuration
in the extension (much like the path to the adapter binary) which will also
involve adding the path as an argument to the adapter